sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -110,7 +264,9 @@ protected BlobClientBase(BlobAsyncClientBase client) {
      * @return a {@link BlobClientBase} used to interact with the specific snapshot.
      */
     public BlobClientBase getSnapshotClient(String snapshot) {
-        return new BlobClientBase(client.getSnapshotClient(snapshot));
+        return new BlobClientBase(this.client, getHttpPipeline(), getAccountUrl(), getServiceVersion(),
```

Comment:
We aren't updating the BlobAsyncClient here which is still being used to download and upload. This is an issue with `getVersionClient`, `getEncryptionScope`, and `getCustomerProvidedKeyClient` as well.